modules/internal/LocaleTree.chpl

```diff
@@ -51,4 +51,51 @@ module LocaleTree {
   }
 
   chpl_initLocaleTree();
+
+  iter localeTree(root = here, parent = here) : locale {
+    halt("DONT USE SERIAL ITER");
+    yield parent;
+  }
+
+  /*
+   Logarithmic gather iterator
+   */
+  iter localeTree(param tag:iterKind, root = here, parent = here) : locale where tag == iterKind.standalone {
```

Comment:
I think we could also rewrite this to take an array of Locales and a ``k`` integer representing the max number of children. At that point we might have to use a coforall instead of a cobegin. That way we wouldn't have to rely on chpl_localeTree.